### PR TITLE
mealie: 1.9.0 -> 1.11.0 + replace relative paths from source with nix store dir

### PR DESCRIPTION
**Pull request by litchipi**
## Description of changes

Context: https://github.com/NixOS/nixpkgs/issues/325391#issuecomment-2257604891
and https://github.com/NixOS/nixpkgs/issues/325120

In the sources, mealie bases all the file hierarchy on `PROJECT_DIR = Path(__file__).parent.parent.parent`.
It breaks everything once the sources are stored in the nix store, which is why we needed the `ALEMBIC_CONFIG_FILE` env var in the first place.

In this PR, in addition to bumping to the latest version, I replaced this with a simple `PROJECT_DIR = Path($out)` line, so that everything works as expected (and removed the `ALEMBIC_CONFIG_FILE` config as we do not need it anymore).

At the same time, as some of the `PROJECT_DIR = ` was left in the `dev/` directory, but this dir is filled with stuff we do not need when serving (hence the name), I simply remove it during patch phase.

This should prevent many future issues on mealie imho

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/mealie/package.nix [hunk 1]
```diff
@@ -92,9 +92,17 @@ pythonpkgs.buildPythonPackage rec {
     uvicorn
   ];
 
-  postPatch = ''
+  patchPhase = ''
```
Comment (ambroisie) [inline pkgs/by-name/me/mealie/package.nix:95]: Keep it as a `postPatch` please, that way someone can still rely on `patches` working when overriding the package.

**Review by ambroisie — CHANGES_REQUESTED** [review]

**Review by ambroisie — APPROVED** [review]
LGTM, wondering whether the choice of putting `alembic.ini` at the root of the package is idiomatic to nixpkgs.

I'll approve, but would recommend asking about that choice on Matrix/Discourse.

**Comment by litchipi** [comment]
@pyrox0 Sorry, you were suggested by Github, so I followed the recommendation :sweat_smile:

**Comment by pyrox0** [comment]
Ah, thank you github :laughing:

**Review by anoadragon453 — APPROVED** [review]
This LGTM, thanks!

I don't have any opinion one way or the other on whether `alembic.ini` should be at the root of the package vs. `/config`. Doing a quick search of nixpkgs for `$out/config` does not reveal many packages using that as a convention. So I don't think it's necessary to put `alembic.ini` under that path.

**Comment by anoadragon453** [comment]
I can also confirm that this patch worked on my own deployment when trying to update to mealie v1.11.0. :white_check_mark:

**Review by anoadragon453 — APPROVED** [review]

**Comment by ambroisie** [comment]
Please squash the update commits and fix the name.

I'll merge once that's done.

**Comment by ambroisie** [comment]
@litchipi sorry, I only wanted the update commits squashed, I think the fix should be in a different commit (atomic commits and all that).

Sorry for the churn.

**Comment by litchipi** [comment]
@ambroisie Yeah no worries that makes sense :sweat_smile:

**Review by dotlambda — COMMENTED** [review]
Commit message doesn't follow contributing guidelines!

**Comment by ambroisie** [comment]
Urgh you're right, sorry I missed it.

**Pull request merged**